llvm/lib/Target/SystemZ/SystemZISelLowering.cpp

```diff
@@ -6829,47 +6803,87 @@ SDValue SystemZTargetLowering::combineMERGE(
   return SDValue();
 }
 
+static bool
+isI128MovedToParts(LoadSDNode *LD,
+                   SmallVector<std::pair<SDNode *, int>, 2> &Users) {
+  // Scan through all users.
+  for (SDNode::use_iterator UI = LD->use_begin(), UIEnd = LD->use_end();
```

Comment:
Unbounded users scans are often a bad idea

Reply:
Good point.  I've changed the logic to ensure early the loop exits early in case of more than two users.